app/src/main/java/com/arctouch/codechallenge/home/HomePresenter.kt

```diff
@@ -0,0 +1,46 @@
+package com.arctouch.codechallenge.home
+
+import android.view.View
+import com.arctouch.codechallenge.api.*
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.data.Repository
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+
+
+class HomePresenter(private val view: HomeInterface.View){
+
+
+    fun onCreateCalled(repository: Repository){
+        val client = Client()
+        val api = client.getClient().create(TmdbApi::class.java)
```

Comment:
I noticed `api` is only used by `repository`. Would it be better if you moved the creation of `api` to be inside the `Repository` class? Are there any downsides to it?

Reply:
It would be a good option, I would see problem only in cases of different uses and that I needed different parameters, as I think that is not the case, moving api into repository would be a good option.